lib/http/2/stream.rb

```diff
@@ -77,6 +77,7 @@ def initialize(connection:, id:, weight: 16, dependency: 0, exclusive: false, pa
       @weight = weight
       @dependency = dependency
       process_priority(weight: weight, stream_dependency: dependency, exclusive: exclusive)
+      @local_window_size = connection.local_settings[:settings_initial_window_size]
```

Comment:
Both @local_window and @local_window_size track "size": one remaining size and other maximum size. Hence my suggestion to rename the latter to @local_window_max. It makes parsing the rest of logic _much_ easier.

Reply:
I'm going to nitpick on top of your nitpick :) and rename it to `local_window_max_size`, as it makes semantically sense (window_max what?).